### PR TITLE
feat: adding sort packages script to package json

### PR DESCRIPTION
**Pull request by evcodes**
#### Description of changes
Allow users to sort their dependencies within any package.json that is modified by running `yarn sort-packages` from the root. This allows them to fix the circle ci check that will be thrown if packages are not sorted correctly. More of a quality of live improvement PR.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [n/a] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [n/a] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [x] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "rm-dev-link": "rimraf -f \"$(yarn global bin)/amplify-dev\"",
     "setup-dev-win": "yarn dev-build && yarn link-win && yarn link-aa-win",
     "setup-dev": "yarn dev-build && yarn rm-dev-link && yarn link-dev && yarn rm-aa-dev-link && yarn link-aa-dev",
+    "sort-packages": "yarn eslint --no-eslintrc --config .eslint.package.json.js **/package.json --fix",
```
Comment (sdstolworthy) [inline package.json:42]:
```suggestion
    "sort-packages": "yarn eslint --no-eslintrc --config .eslint.package.json.js **/package.json --fix || echo 'The dependencies in `package.json` are not sorted correctly, please run ...'",
```

**Review by goldbez — APPROVED** [review]

**Review by sdstolworthy — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -39,6 +39,7 @@
     "rm-dev-link": "rimraf -f \"$(yarn global bin)/amplify-dev\"",
     "setup-dev-win": "yarn dev-build && yarn link-win && yarn link-aa-win",
     "setup-dev": "yarn dev-build && yarn rm-dev-link && yarn link-dev && yarn rm-aa-dev-link && yarn link-aa-dev",
+    "sort-packages": "yarn eslint --no-eslintrc --config .eslint.package.json.js **/package.json --fix || echo 'The dependencies in `package.json` are not sorted correctly, please run yarn sort-packages to fix this issue.'",
```
Comment (sobolk) [inline package.json:42]: If we're passing `--fix` why would we `echo` ? Should echo actually go to `lint-check-package-json` here https://github.com/aws-amplify/amplify-cli/blob/d0ffcc53112efc22d138bf52eecd40e37f98ce8d/package.json#L30

Comment (evcodes) [reply package.json:42]: This is what I was intending to happen. When a user commits something that is not correct, have the error message guide them in CI.

Comment (evcodes) [reply package.json:42]: I will address this in a new PR.

**Pull request merged**